src/Presentation/FacetUiBuilder.php

```diff
@@ -39,40 +44,46 @@ public function __construct(
 	// TODO: parameter: selected values (from QueryStringParser https://github.com/ProfessionalWiki/WikibaseFacetedSearch/issues/31)
 	public function createHtml( ItemId $itemType ): string {
 		$this->config->getFacetConfigForInstanceType( $itemType );
-		$this->query = $this->getSearchQueryFromUrl();
+
+		$this->query = ( new QueryStringParser() )->parse( $this->getSearchQueryFromUrl()['search'] );
+		$this->constraints = $this->query->getConstraintsPerProperty();
```

Comment:
$this->query is unused in the class, it could be a local variable.

You could compute `$this->constraints` in its own method. Then in this one you would just have

`$this->constraints = $this->getConstraintsPerProperty()`

... there are other issues, though. Unsure if when resolving all you would have such code.